swarm/storage/localstore.go

```diff
@@ -184,3 +184,40 @@ func (ls *LocalStore) Iterator(from uint64, to uint64, po uint8, f func(Address,
 func (ls *LocalStore) Close() {
 	ls.DbStore.Close()
 }
+
+func (ls *LocalStore) Migrate() error {
+	schema, err := ls.DbStore.GetSchema()
+	if err != nil {
+		log.Error(err.Error())
+		return err
+	}
+
+	log.Debug("found schema", "schema", schema, "runtime-schema", CurrentDbSchema)
+	if schema != CurrentDbSchema {
+		// run migrations
+
+		if schema == "" {
```

Comment:
@nolash this is the relevant piece of code, that I want you and @jpeletier to be aware of.

The idea is that once we release this, every node that runs swarm will have a `schema == ""`, so we will run `cleanupFunc` over every single chunk in the LevelDB.

So if none of the validators pass (which currently is validators for MRU, and validator for Content Addressed chunks), then we remove the chunk.

Then we bump up the schema to the first named schema (to be decided), and we try not to store any more invalid chunks :)

I hope that makes sense.

Reply:
I think this makes sense.